src/gpu/intel/ocl/usm_utils.cpp

```diff
@@ -57,11 +57,17 @@ void *malloc_host(impl::engine_t *engine, size_t size) {
             size_t, cl_uint, cl_int *);
 
     if (size == 0) return nullptr;
+    bool large_buffer = size
+            > utils::downcast<const xpu::ocl::engine_impl_t *>(engine->impl())
+                      ->max_allocation_size();
+    static cl_bitfield properties[]
```

Comment:
Minor: I'm not sure `static` is good for `properties` object that's passed to CL calls especially in case it will need to be extended in the future. Also current code doesn't provide a scalable way of expanding it either, probably a TODO with that note could help to look into that direction when/if needed.